### ISSUE TITLE
How to calculate top-k in-class accuracies using TensorFlow Slim metrics

### ISSUE THREAD

**Issue by Johnson145** [tensorflow/models#3344]
I already asked this question on [stackoverflow](https://stackoverflow.com/questions/48600929/how-to-calculate-top-k-in-class-accuracies-using-tensorflow-slim-metrics), but didn't get an answer. So I hope it's okay to repost my question here:

I'd like to extend [this script][1] such that it is able to evaluate the top-k accuracies per class. I hope it boils down to adding a metric to the following code snippet:

    # Define the metrics:
    names_to_values, names_to_updates = slim.metrics.aggregate_metric_map({
        'Accuracy': slim.metrics.streaming_accuracy(predictions, labels),
        'Recall_5': slim.metrics.streaming_recall_at_k(
            logits, labels, 5), })

I already followed [this comment][2] to add the confusion matrix, which allows me to calculate the top1 in-class accuracies. However, I'm not sure how to get the top-k values as I can't find an appropriate slim metric.

To clarify:

 - I'm not looking for the average top-k accuracy, but the per class values.
 - I'm able to implement the required calculations using basic tensors, but I'm new to the slim interface and don't know how to do it using the above script.

  [1]: https://github.com/tensorflow/models/blob/master/research/slim/eval_image_classifier.py
  [2]: https://github.com/tensorflow/models/issues/1286#issuecomment-317205632

**Comment by michaelisard** [comment]
I'm sorry, but I think your best bet is to keep looking for answers on stackoverflow. These issues are only for bug reports and feature requests, and don't get as many eyes on them in any case.
